packages/uiweb/src/lib/hooks/chat/usePushChatStream.ts

```diff
@@ -0,0 +1,123 @@
+    const [chatRequestStream, setChatRequestStream] = useState<any>({}); // any message in request
+    const [groupMetaStream, setGroupMetaStream] = useState<any>({}); //group info
+
+    const addSocketEvents = async () => {
+        console.warn('\n--> addChatSocketEvents - stream');
+        pushChatStream?.on(CONSTANTS.STREAM.CONNECT, (err: Error) => {
+            console.log('CONNECTED - stream: ', err);
+            setIsPushChatStreamConnected(true);
+        });
+      
+        pushChatStream?.on(CONSTANTS.STREAM.DISCONNECT, (err: Error) => {
+            console.log('DIS-CONNECTED: - stream ', err);
+            setIsPushChatStreamConnected(false);
+        });
+
+
+        //Listen for chat messages, your message, request, accept, rejected,
+        console.log('\t-->will attach eachMessage event now - stream');
+        pushChatStream?.on(CONSTANTS.STREAM.CHAT, (message: any) => {
+
+            if ((message.event === "chat.request")) {
+                setChatRequestStream(message);
+            } else {
+                setChatStream(message);
+            }
+
+        });
+        pushChatStream?.on(CONSTANTS.STREAM.CHAT_OPS, (chatops: any) => {
+            setGroupMetaStream(chatops)
+        });
+
+    };
+
+
+
+    const removeSocketEvents = () => {
+        pushChatStream?.disconnect();
+    };
+
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    useEffect(() => {
+        if (pushChatStream) {
+            addSocketEvents();
+        }
+
+        return () => {
+            if (pushChatStream) {
+                removeSocketEvents();
+            }
+        }
+    }, [pushChatStream]);
+
+   
+    /**
+     * Whenever the requisite params to create a connection object change
+     *  - disconnect the old connection 
+     *  - create a new connection object
+     */
+
+    useEffect(() => {
+        if (pushUser) {
+            if(!pushChatStream) {
+                console.log(pushChatStream)
+                const main = async () => {
+                    const newstream = await pushUser?.initStream(
+                        [
+                            CONSTANTS.STREAM.CHAT,
+                            CONSTANTS.STREAM.CHAT_OPS,
+                            CONSTANTS.STREAM.CONNECT,
+                            CONSTANTS.STREAM.DISCONNECT,
+
+                        ],
+                        {
+
+                            connection: {
+                                retries: 3, // number of retries in case of error
+                            },
+                            raw: true   
+                        }
+                    );
+                    console.log('new connection object: ---- ', newstream);
+                    await newstream?.connect();
+                    setPushChatStream(newstream);
+
+                };
+                main().catch((err) =>
+                    console.log("error initializing the stream", err)
+                );
```

Comment:
We can remove any initialisation that might already exist as if the chatid constantly changes:
``` return () => {
                    if(pushChatStream){
                        pushChatStream?.disconnect?.();
                    }
                    
                }; ```